marlowe-integration-tests/test/Language/Marlowe/Runtime/Integration/MarloweQuery.hs

```diff
@@ -96,21 +107,87 @@ instance PaginatedQuery GetHeaders where
     Contract2 -> standardContractHeader contract2
     Contract3 -> standardContractHeader contract3
     Contract4 -> standardContractHeader contract4
-  toFilter _ MarloweQueryTestData{..} ContractFilterSym{..} =
+  toFilter _ testData@MarloweQueryTestData{..} ContractFilterSym{..} =
     ContractFilter
       { tags = testTagsToTags tagsSym
       , roleCurrencies = flip Set.map roleCurrenciesSym \case
           Contract1 -> standardContractRoleCurrency contract1
           Contract2 -> standardContractRoleCurrency contract2
           Contract3 -> standardContractRoleCurrency contract3
           Contract4 -> standardContractRoleCurrency contract4
+      , partyAddresses = partyAddressesToAddresses testData partyAddressesSym
+      , partyRoles =
+          flip Set.map partyRolesSym $
+            flip AssetId "Party A" . \case
+              Contract1 -> standardContractRoleCurrency contract1
+              Contract2 -> standardContractRoleCurrency contract2
+              Contract3 -> standardContractRoleCurrency contract3
+              Contract4 -> standardContractRoleCurrency contract4
       }
-  enumerateFilters q = do
-    tagsSym <- Set.toList $ Set.powerSet $ Set.fromList [minBound .. maxBound]
-    roleCurrenciesSym <- Set.toList $ Set.powerSet $ Set.fromList $ allRefs q
-    pure ContractFilterSym{..}
+  enumerateFilters q =
+    nub $
+      defaultContractFilter
+        : fold
+          [ do
+              tagsSym <- Set.toList $ Set.powerSet $ Set.fromList [minBound .. maxBound]
+              filterVariations defaultContractFilter{tagsSym}
+          , do
+              roleCurrenciesSym <- Set.toList $ Set.powerSet $ Set.fromList $ allRefs q
+              filterVariations defaultContractFilter{roleCurrenciesSym}
+          , do
+              partyAddressesSym <- Set.toList $ Set.powerSet $ Set.fromList [minBound .. maxBound]
+              filterVariations defaultContractFilter{partyAddressesSym}
+          , do
+              partyRolesSym <- Set.toList $ Set.powerSet $ Set.fromList $ allRefs q
+              filterVariations defaultContractFilter{partyRolesSym}
```

Comment:
I had to change the method of enumerating the filters, as the nested power sets were making the test suite way too huge. Even this method produces ~10,000 test cases, but continuing with the nested power sets lead to >100,000 test cases.